#! /bin/sh /usr/share/dpatch/dpatch-run
## 43_scripts__mysql_update__password.dpatch by <ch@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Adds --password option needed for /etc/mysql/debian-start.
## DP: http://bugs.mysql.com/bug.php?id=19400

@DPATCH@

--- old/scripts/mysql_upgrade_shell.sh.ch	2006-04-27 21:54:24.000000000 +0200
+++ new/scripts/mysql_upgrade_shell.sh	2006-04-27 21:54:55.000000000 +0200
@@ -14,6 +14,7 @@
 
 
 user=root
+password=''
 
 case "$1" in
     --no-defaults|--defaults-file=*|--defaults-extra-file=*)
@@ -36,6 +37,7 @@
     case "$arg" in
       --basedir=*) MY_BASEDIR_VERSION=`echo "$arg" | sed -e 's/^[^=]*=//'` ;;
       --user=*) user=`echo "$arg" | sed -e 's/^[^=]*=//'` ;;
+      --password=*) password=`echo "$arg" | sed -e 's/^[^=]*=//'` ;;
       --ldata=*|--data=*|--datadir=*) DATADIR=`echo "$arg" | sed -e 's/^[^=]*=//'` ;;
       --force) force=1 ;;
       --verbose) verbose=1 ;;
@@ -100,6 +103,7 @@
   echo "  --datadir  Specifies the data directory"
   echo "  --force    Mysql_upgrade.info file will be ignored"
   echo "  --user     Username for server login if not current user"
+  echo "  --password Password for server login if not empty"
   echo "  --verbose  Display more output about the process"
   echo ""
 
@@ -179,7 +179,7 @@
     then
        echo "mysql_upgrade already done for this version"
     fi
-    $bindir/mysql_fix_privilege_tables --silent $args
+    $bindir/mysql_fix_privilege_tables --silent --user=$user --password=$password $args
     exit 0
   fi
 fi
@@ -185,7 +189,7 @@
 # Run the upgrade
 #
 
-check_args="--check-upgrade --all-databases --auto-repair --user=$user"
+check_args="--check-upgrade --all-databases --auto-repair --user=$user --password=$password"
 
 if test $verbose = 1
 then
@@ -200,4 +204,4 @@
   echo "@MYSQL_BASE_VERSION@" > $CHECK_FILE
 fi
 
-$bindir/mysql_fix_privilege_tables --silent --user=$user $args
+$bindir/mysql_fix_privilege_tables --silent --user=$user --password=$password $args
